### ISSUE TITLE
Explicit validator in parent class breaks type validation in child class for nested types

### ISSUE THREAD

**Issue by AlbertMukhammadiev** [samuelcolvin/pydantic#1015]
# Question

Please complete:
* OS: Ubuntu 18.04
* Python version `import sys; print(sys.version)`: 3.6.8 [GCC 7.4.0]
* Pydantic version `import pydantic; print(pydantic.VERSION)`: 1.2a1

Hello, I'm trying to create an instance of class Bar.
```py
import pydantic
from typing import List


class Foo(pydantic.BaseModel):
    foo: List[List[int]]

    @pydantic.validator('foo')
    def check_something(cls, value):
        return value


class Bar(Foo):
    pass


foo_ = [[0, 1]]
Foo(foo=foo_)
Bar(foo=foo_)
```

Output:
```
pydantic.error_wrappers.ValidationError: 1 validation error for Bar
foo -> 0
  value is not a valid integer (type=type_error.integer)
```
There will be no error, If you remove validator.

**Comment by samuelcolvin** [comment]
Confirmed, thanks for reporting. Has been this way since v1.

I'll try and fix forthwith.

**Comment by samuelcolvin** [comment]
should be fixed via #1016, please let me know if that solves your problem. If so I'll release it as `v1.1.1`

**Comment by AlbertMukhammadiev** [comment]
Great! Thank you for the quick response!
